webapp/app/forms/fields.py

```diff
@@ -25,9 +25,18 @@ def _add_classes_to_kwargs(kwargs, classes):
 
 class NumericInputMixin:
 
-    @staticmethod
-    def set_inputmode(kwargs):
+    @classmethod
+    def set_inputmode(cls, kwargs):
         kwargs.setdefault('inputmode', 'numeric')
+
+        return kwargs
+
+
+class NumericInputMaskMixin(NumericInputMixin):
+
+    @classmethod
+    def set_inputmode(cls, kwargs):
```

Comment:
Why `classmethod` not `@staticmethod`? The methods don't use the class.

Reply:
Inheritance only works with class methods and not with static methods. We use the class implicit in the `super()` call.